server/channels/api4/channel_bookmark.go

```diff
@@ -18,6 +18,7 @@ func (api *API) InitChannelBookmarks() {
 	api.BaseRoutes.ChannelBookmark.Handle("/sort_order", api.APISessionRequired(updateChannelBookmarkSortOrder)).Methods("POST")
 	api.BaseRoutes.ChannelBookmark.Handle("", api.APISessionRequired(deleteChannelBookmark)).Methods("DELETE")
 	api.BaseRoutes.ChannelBookmarks.Handle("", api.APISessionRequired(listChannelBookmarksForChannel)).Methods("GET")
+	api.BaseRoutes.ChannelBookmarks.Handle("/open_graph", api.APISessionRequired(fetchOpenGraph)).Methods("GET")
```

Comment:
We had this before but removed it because of security issues. There needs to be a thorough discussion of what needs to be done in order to reintroduce it. And if/when we do reintroduce it, I'd prefer we use the original implementation (with required improvements) instead of implementing a new API just for this use case specifically.

https://github.com/mattermost/mattermost/pull/24177